File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/Application.java
package com.github.arhor.simple.expense.tracker;

import lombok.extern.slf4j.Slf4j;
import lombok.val;

import org.springframework.boot.ApplicationRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
import org.springframework.context.annotation.Bean;
import org.springframework.core.env.Environment;

import com.github.arhor.simple.expense.tracker.config.properties.ApplicationProps;

@Slf4j
@SpringBootApplication(proxyBeanMethods = false)
@ConfigurationPropertiesScan(basePackageClasses = ApplicationProps.class)
public class Application {

    public static void main(final String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Bean
    public ApplicationRunner applicationRunner(final Environment env) {
        return args -> {
            val port = env.getProperty("server.port");
            val path = env.getProperty("server.servlet.context-path", "/");

            if ((port != null) && (path != null)) {
                log.info("Local access URL: http://localhost:{}{}", port, path);
            }
        };
    }
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/Application.kt
package com.github.arhor.simple.expense.tracker

import com.github.arhor.simple.expense.tracker.config.props.ApplicationProps
import org.slf4j.LoggerFactory
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.boot.web.context.WebServerApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Profile
import org.springframework.web.context.WebApplicationContext
import java.lang.invoke.MethodHandles

private val log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass())

@SpringBootApplication(proxyBeanMethods = false)
@ConfigurationPropertiesScan(basePackageClasses = [ApplicationProps::class])
class Application {

    @Bean
    @Profile("dev")
    fun <T> displayApplicationInfo(context: T)
        where T : WebApplicationContext,
              T : WebServerApplicationContext = ApplicationRunner {

        val port = context.webServer.port
        val path = context.servletContext?.contextPath ?: ""

        log.info("Local access URL: http://localhost:{}{}", port, path)
    }
}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
